### DISCUSSION TITLE
Flight envelope

### DISCUSSION THREAD

**Discussion by Judedarwinism** [JSBSim-Team/jsbsim#752] [Q&A]
Can i obtain the explicit description of flight envelope of aircraft model in JSBSim?

**Comment by seanmcleod** [comment]
No, the data in the aircraft model file doesn't include that explicitly.

**Comment by bomber64** [comment]
When you say flight envelope....

Are you after flying the plane and  recording how it performs ?

If so that is possible

Or is it something else you're after ?

**Reply by agodemar** [reply]
@bomber64 'Flight envelope' is not an ambiguous term.
See here: https://en.wikipedia.org/wiki/Flight_envelope
Probably @Judedarwinism refers to the altitude envelope, or 'H-M diagram'
![immagine](https://user-images.githubusercontent.com/819499/199468566-6ab49325-487a-4c1f-8690-cb4e0916bddb.png)

**Reply by Judedarwinism** [reply]
Yes, that's what i mean in this picture. Thanks very much. To more precisely speak, i want to know and quantify the boundary conditions of the airplane in JSBSim.

**Reply by seanmcleod** [reply]
@Judedarwinism were you hoping to open a JSBSim aircraft model file, e.g. https://github.com/JSBSim-Team/jsbsim/blob/master/aircraft/737/737.xml and find these sorts of flight envelopes listed? 

Or are you asking whether there is some JSBSim function which you can use to load a particular aircraft model file and then have JSBSim generate these flight envelopes for you?

Neither option, i.e. pre-generated data in the aircraft model file or some JSBSim envelope function are available.

So you'll either need to fly the model as a test pilot to collect the relevant data, or write an auto-pilot to act as a test pilot, or make use of some of the data you find in the aircraft model file and perform your own calculations. For example that's what I do with my EM Diagram generator - https://github.com/seanmcleod/EMDiagrams/blob/master/EMDiagrams.ipynb

You could also make use of say some python scripting with JSBSim to calculate some of the flight envelope boundaries.
